typify-impl/src/value.rs

```diff
@@ -524,14 +551,160 @@ mod tests {
+
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(25))
+                .map(|x| x.to_string()),
+            Some("25_i8".to_string()),
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(20))
+                .map(|x| x.to_string()),
+            Some("20_i8".to_string()),
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(30))
+                .map(|x| x.to_string()),
+            Some("30_i8".to_string()),
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(26))
+                .map(|x| x.to_string()),
+            None,
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(19))
+                .map(|x| x.to_string()),
+            None,
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(31))
+                .map(|x| x.to_string()),
+            None,
+        );
+    }
+
+    #[test]
+    fn test_value_validate_numbers() {
+        let mut schema = schema_for!(f64);
+        let validation = Some(Box::new(NumberValidation {
+            multiple_of: Some(5.0),
+            minimum: Some(20.0),
+            maximum: Some(30.0),
+            exclusive_maximum: None,
+            exclusive_minimum: None,
+        }));
+        drop(schema.schema.number);
+        schema.schema.number = validation;
+
+        let type_name = type_name::<f64>();
+        let name = type_name
+            .rsplit_once("::")
+            .map_or(type_name, |split| split.1)
+            .to_string();
+
+        let mut type_space = TypeSpace::default();
+
+        // Convert all references
+        type_space
+            .add_ref_types(schema.definitions.clone())
+            .unwrap();
+        let type_id = if let Some(already_type_id) = type_space.ref_to_id.get(&name) {
+            already_type_id.clone()
+        } else {
+            type_space
+                .add_type_with_name(&schema.schema.into(), Some(name.to_string()))
+                .unwrap()
+        };
+
         let type_entry = type_space.id_to_entry.get(&type_id).unwrap();
 
         assert_eq!(
             type_entry
-                .output_value(&type_space, &json!([0, 0, 0, "zero"]))
+                .output_value(&type_space, &json!(25.0))
+                .map(|x| x.to_string()),
+            Some("25.0_f32".to_string()),
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(20.0))
+                .map(|x| x.to_string()),
+            Some("20.0_f32".to_string()),
+        );
+        assert_eq!(
+            type_entry
+                .output_value(&type_space, &json!(30.0))
+                .map(|x| x.to_string()),
+            Some("30.0_f32".to_string()),
```

Comment:
Seems that we are moving from `f64` to `f32` since the min and max are within the bounds of `f32`. However, we will lose precision, so this might be a bug? Not sure what the intended behavior is. Could you weight in @ahl?